oslo_vmware/rw_handles.py

```diff
@@ -297,9 +298,13 @@ def __init__(self, session, lease, url, file_handle,
         self._url = url
         self._last_logged_progress = 0
         self._last_progress_udpate = 0
+        self._last_progress = 0
+        self._last_progress_change = 0
         self._updater = None
 
         if update_progress:
+            self._last_progress_change = self._last_progress_update = \
+                time.time()
```

Comment:
We need to initialize those 2 variables properly. Nova doesn't set `update_progress` but still calls `update_progress()` itself. Can we please move this initialization up or could you explain to me, why we should only set them if `update_progress` is set?

The only reason I can see is that we would want to have "never updated" as a distinct state represented in the variables being `0`. But even then: we update them here before the first run of `self..update_progress()`, so I don't see us reaching that goal with the current code either.

Reply:
Ok I pulled out the init of those 2 vars from the if update_progress: